yaml-cpp vendor fails to build with newest VS2017 15.8

Problem: yaml-cpp won't build with newest Visual Studio. Error:
```
<...>\yaml-cpp\exceptions.h(149): error C3646: '_NOEXCEPT': unknown override specifier (compiling source file <...>\src\parse.cpp) [<...>\yaml-cpp.vcxproj]
```

There is already a fix upstream, but it's not released, yet:
https://github.com/jbeder/yaml-cpp/commit/0f9a586ca1dc29c2ecb8dd715a315b93e3f40f79

Either we have to apply the fix ourselves or wait for the next 0.6 release and upgrade.

--- a/rviz_yaml_cpp_vendor/CMakeLists.txt
+++ b/rviz_yaml_cpp_vendor/CMakeLists.txt
@@ -31,9 +31,11 @@ macro(build_yaml_cpp)
   # This specific version (past the current latest release of 0.5.3) is required to make
   # yaml-cpp relocatable, hopefully it is released again soon.
   # See: https://github.com/jbeder/yaml-cpp/pull/538
-  ExternalProject_Add(yaml_cpp-86ae3a5
-    URL https://github.com/jbeder/yaml-cpp/archive/86ae3a5aa7e2109d849b2df89176d6432a35265d.zip
-    URL_MD5 28091f7f66086a84b14da1f5d798c6a6
+  # Latest release fails to compile on recent visual studio (VS2017 v15.8.1)
+  # See: https://github.com/jbeder/yaml-cpp/pull/597
+  ExternalProject_Add(yaml_cpp-0f9a586
+    URL https://github.com/jbeder/yaml-cpp/archive/0f9a586ca1dc29c2ecb8dd715a315b93e3f40f79.zip
+    URL_MD5 ec76c27ebd07d5178cbe85b773df8e62
     TIMEOUT 600
     LOG_CONFIGURE ${should_log}
     LOG_BUILD ${should_log}

